lib/rules/no-constant-condition.js

```diff
@@ -215,6 +216,13 @@ module.exports = {
                     return isConstant(node.expressions[node.expressions.length - 1], inBooleanPosition);
                 case "SpreadElement":
                     return isConstant(node.argument, inBooleanPosition);
+                case "CallExpression":
+                    if (node.callee.type === "Identifier" && node.callee.name === "Boolean") {
+                        return node.arguments.length === 0 || isConstant(node.arguments[0], true);
+                    }
+                    return false;
+                case "Identifier":
+                    return node.name === "undefined";
```

Comment:
Can we also check if these identifiers are references to global variables `Boolean` and `undefined`, to avoid false positives when those globals are shadowed?

```js
function foo(Boolean, undefined) {

    if (Boolean()); // false positive

    if (undefined); // false positive

}

```

We could use `context.getScope()` to get the scope where the currently traversed node is, then find the reference in [`scope.references`](https://eslint.org/docs/developer-guide/scope-manager-interface#references) ([`reference.identifier`](https://eslint.org/docs/developer-guide/scope-manager-interface#identifier) should be the identifier node we're checking) and check if `reference.resolved.scope.type === "global"`.

Reply:
Hmm. When I do this, I get `"global"` when evaluating:

```
const Boolean = () => {};
if (Boolean(1)) {}
```

Would it work to check `reference.resolved.defs.length === 0`?

Reply:
> Would it work to check `reference.resolved.defs.length === 0`

Good point! Yes, we should definitely add that condition.